jerry-ext/include/jerryscript-ext/module.h

```diff
@@ -83,6 +83,12 @@ typedef struct jerryx_native_module_t
   {                                                                    \
     jerryx_native_module_unregister(&_##module_name##_definition);     \
   }
+  
+#define JERRYX_NATIVE_MODULE_EXTERN(module_name) \
+  extern jerryx_native_module_t _ ## module_name ## _definition
```

Comment:
IINM this is insufficient, because `_ ## module_name ## _definition` is declared static, so declaring it extern is not going to help find the symbol.

You may want something like
```C
#define JERRYX_NATIVE_MODULE_STRUCTURE(qualifier, module_name, on_resolve_cb) \
  qualifier jerryx_native_module_t _ ## module_name ## _definition =      \
  {                                                                    \
    .name_p = (jerry_char_t *) #module_name,                           \
    .on_resolve_p = (on_resolve_cb),                                   \
    .next_p = NULL                                                     \
};
```

and then start `JERRYX_NATIVE_MODULE_IMPLEM()` with
```C
  JERRYX_NATIVE_MODULE_STRUCTURE(static, module_name, (on_resolve_cb))
  ...
```

You'll probably want to store the rest of the definition of `JERRYX_NATIVE_MODULE_IMPLEM()` in its own macro, like `JERRYX_NATIVE_MODULE_ENTRY_POINTS()` and then define `JERRYX_NATIVE_MODULE_EXTERN()` as
```C
#define JERRYX_NATIVE_MODULE_EXTERN(module_name, on_resolve_cb) \
  JERRYX_NATIVE_MODULE_EXTERN_IMPLEM(module_name, (on_resolve_cb))
#define JERRYX_NATIVE_MODULE_EXTERN_IMPLEM(module_name, on_resolve_cb) \
  JERRYX_NATIVE_MODULE_STRUCTURE(, module_name, on_resolve_cb) \
  JERRYX_NATIVE_MODULE_ENTRY_POINTS(module_name)
```

Reply:
I tested with the following:
```c
// top.h
extern int i;

// left.c
static int i = 0;

// right.c
int main() { return i; }
```

And as you might have expected, it did fail. And later, I learned why: https://stackoverflow.com/a/2841771/2423150

```
$ gcc left.c right.c -o middle
left.c:3:12: error: static declaration of 'i' follows non-static declaration
static int i = 0;
           ^
./top.h:1:12: note: previous declaration is here
extern int i;
           ^
1 error generated.
```

So using `static` in the context of this structure is very likely a bad idea... What do you think about using `const`, though?

Reply:
@gabrielschulhof I made a second attempt at the macros. Hope this one is better! :)